server/banned.go

```diff
@@ -32,46 +34,48 @@ func (pi *packetInfo) checkBlameMessage() error {
 		return nil
 	}
 
+	reason := packet.GetMessage().GetBlame().GetReason()
 	validBlame := false
 
-	for _, reason := range validBlamereasons {
-		if packet.GetMessage().GetBlame().GetReason() == reason {
+	for _, r := range validBlamereasons {
+		if reason == r {
 			validBlame = true
+			break
 		}
 	}
 
-	if validBlame {
+	if !validBlame {
+		return fmt.Errorf("unknown blame reason: %s", reason)
```

Comment:
I think this is fine, but it technically is a logic change. This now errors if the blame isn't valid, before it just ignored it.

Reply:
Yeah I should have done all of the logic changes in separate PRs. That is really my bad.

Would you like me to do that? I could extract them and try to isolate this PR to only logging.

Reply:
I think the changes included are ok for now. We have already gone through them in detail. =)